src/test/java/io/rcktapp/api/handler/sql/SqlGetHandlerTest.java

```diff
@@ -0,0 +1,191 @@
+        String[] inClause = {"updatexml(rand()", "concat(CHAR(126)", "user()", "CHAR(126))", "null)"};
+
+        // Setup request object and sql table objects to properly reflect
+        Request request = new Request(url, "GET", new HashMap<>(), new HashMap<>(), "");
+        Service.ApiMatch apiMatch = new Service.ApiMatch(api, endpoint, "GET", request.getUrl(), "http://localhost:8080/api/lift/us/", collectionName + "/" + passedQuery + "/");
+        Collection collection = new Collection();
+        Entity entity = new Entity(new Table(sqlDb, collectionName));
+        Attribute attribute = new Attribute();
+
+        attribute.setName(collectionName);
+        entity.setKey(attribute);
+        collection.setEntity(entity);
+        request.setApiMatch(apiMatch);
+
+        // Mocked methods that return local objects used in the future
+        when(chain.getService()).thenReturn(service);
+        when(service.getDb(request.getApi(), request.getCollectionKey(), SqlDb.class)).thenReturn(sqlDb);
+        when(sqlDb.getConnection()).thenReturn(connection);
+
+        try (MockedStatic<Rql> rqlMockedStatic = mockStatic(Rql.class)) {
+            try (MockedStatic<Sql> sqlMockedStatic = mockStatic(Sql.class)) {
+
+                rqlMockedStatic.when(() -> Rql.getRql(sqlDb.getType())).thenReturn(sqlRql);
+
+                when(api.getCollection(request.getCollectionKey(), SqlDb.class)).thenReturn(collection);
+                sqlMockedStatic.when(() -> Sql.getInClauseStr(J.explode(",", Sql.check(request.getEntityKey())))).thenReturn(passedQuery);
+
+                // When we call the service method on the mocked handler actually call the real method
+                doCallRealMethod().when(handler).service(service, api, endpoint, action, chain, request, response);
+
+                // More mocked methods that return objects reused locally
+                when(sqlRql.asCol("plugins")).thenReturn(collectionName);
+                sqlMockedStatic.when(() -> Sql.check(collectionName)).thenReturn(collectionName);
+                sqlMockedStatic.when(() -> Sql.getQuestionMarkStr(inClause)).thenReturn("?,?,?,?,?");
+                when(sqlRql.createStmt(any(), any(), any(), any(Replacer.class))).thenReturn(stmt);
+                when(chain.getConfig("maxRows", 0)).thenReturn(0);
+
+                // Call the actual method on the mocked handler
+                handler.service(service, api, endpoint, action, chain, request, response);
+
+                // Verify that the passed in query contains no SQL functions
+                verify(sqlRql).createStmt(sqlCaptor.capture(), any(), rqlParamsCaptor.capture(), any(Replacer.class));
+                assertEquals(expectedQuery, sqlCaptor.getValue());
+                // Assert that we are passing in the correct rql parameters parsed from the URL
+                assertEquals(request.getParams(), rqlParamsCaptor.getValue());
```

Comment:
Is there any difference between this test and the test above other than this line? If not, we probably don't need both these tests

Although it would be nice to see a test that hits this code that is not attempting to do SQL injection (like the `updatexml` query is doing)

Reply:
Yeah there is basically no difference.

Reply:
Also a regular query I don't think would be testing the specific areas of the code that were changed.